lib/index.js

```diff
@@ -136,18 +127,6 @@ const pkgContents = async ({
 
   const recursePromises = []
 
-  // if we didn't get withFileTypes support, tack that on
-  if (typeof dirEntries[0] === 'string') {
-    // use a map so we can return a promise, but we mutate dirEntries in place
-    // this is much slower than getting the entries from the readdir call,
-    // but polyfills support for node versions before 10.10
-    await Promise.all(dirEntries.map(async (name, index) => {
-      const p = resolve(path, name)
-      const st = await lstat(p)
-      dirEntries[index] = Object.assign(st, { name })
-    }))
-  }
```

Comment:
This drops support for Node without `withFileTypes` support (before 10.10). That's outside of the engines of this repo, but should this be a breaking change?

Reply:
Nope. This is why we have engines: so we can declare the compatibility first and then make the changes later.  This would be no different than if we introduced something like static private methods in a fix commit.